src/services/spectral-agent-service.ts

```diff
@@ -96,12 +96,19 @@ export class SpectralAgentService {
   }
 
   public processResults(results: ScanResult, folderPath: string) {
-    this.resetFindings()
     results?.items?.forEach((item: ScanFindingView) => {
       this.processFindingItem({ item, folderPath: folderPath.toLowerCase() })
     })
   }
 
+  public resetFindings() {
+    this.findings = { [FindingType.iac]: {}, [FindingType.secret]: {} }
+    this.findingsAggregations = {
+      [FindingType.secret]: 0,
```

Comment:
consider giving a `Record<FindingType, number>` type for the dicts. it will enforce the usage of all finding types (whenever we will add oss, or other future types)

Reply:
`this.findings` is of type `Findings`, which eventually is `export type Findings = Record<FindingType, FindingsTypeResults>`

Reply:
fixed https://github.com/SpectralOps/vscode-extension/pull/8/commits/fffbca1c6489b308e5c02704e1ce97ef8d208ff4